src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -724,6 +805,332 @@ public final void markDirty()
     }
 
     /**
+     * Add a new container to the building.
+     * @param pos position to add.
+     */
+    public void addContainerPosition(BlockPos pos)
+    {
+        containerList.add(pos);
+    }
+
+    /**
+     * Remove a container from the building.
+     * @param pos position to remove.
+     */
+    public void removeContainerPosition(BlockPos pos)
+    {
+        containerList.remove(pos);
+    }
+
+    /**
+     * Get all additional containers which belong to the building.
+     * @return a unmodifiable list of the container positions.
+     */
+    public List<BlockPos> getAdditionalCountainers()
+    {
+        return new ArrayList<>(containerList);
```

Comment:
The javadoc says we are returning an unmodifiable list? So with that exception, what is modifying it?